databaseConnect/index.js

```diff
@@ -0,0 +1,58 @@
+const pg = require("pg");
+
+const { Client } = pg;
+const client = new Client({
+    user: "postgres",
+    password: "2020327",
+    host: "localhost",
+    port: "5432",
+    database: "test",
+});
+
+const init = () => {
+    return client.connect();
+};
+
+const query = (text, params, callback) => {
+    return client.query(text, params, callback);
+};
+
+const getUserByUsername = (username, callback) => {
+    return query(
+        "SELECT * FROM users WHERE username = $1",
+        [username],
+        callback
+    );
+};
+
+const setUser = (
+    { username, password, firstName, lastName, age, salt },
+    callback
+) => {
+    return query(
+        "INSERT INTO users(username, password, firstname, lastname, age , salt) VALUES ($1, $2, $3, $4, $5, $6)",
+        [username, password, firstName, lastName, age, salt],
+        callback
+    );
+};
+
+const getProjects = (callback) => {
+    return query("SELECT * FROM project", null, callback);
```

Comment:
why passing null?

Reply:
Since we pass the callback as the 3rd parameter, without the 2nd parameter, null is used instead.
https://node-postgres.com/features/queries

Reply:
we don't need this method at all, as mentioned higher

Reply:
Otherwise i have an issue with empty query response.
![Screenshot_7](https://github.com/user-attachments/assets/26d1a854-b8c7-4c9a-a82d-18070f689829)
![Screenshot_9](https://github.com/user-attachments/assets/11fbc73b-5460-4a83-9043-d84c786b989b)
![Screenshot_10](https://github.com/user-attachments/assets/97957c43-7bff-4ae7-baa5-74051f7a0004)